crates/wiggle/generate/src/funcs.rs

```diff
@@ -12,6 +12,12 @@ pub fn define_func(
 ) -> TokenStream {
     let funcname = func.name.as_str();
 
+    // `proc_exit` is special; it's essentially an unwinding primitive,
+    // so we implement it in the runtime rather than in wasi-common.
+    if funcname == "proc_exit" {
+        return quote!();
+    }
```

Comment:
This would break Lucet, where we deal with `proc_exit` in a totally different manner. We install a panic handler at the wasm->host boundary that terminates the instance, and then just `panic!` in our proc_exit trait method.

If that doesn't work for wasmtime, can we figure out a way to deal with `noreturn` functions in wiggle that works for both runtimes?

Reply:
I considered using panics here, but didn't because of the possibility of users embedding Wasmtime in programs built with `panic = "abort"`.

My main idea is that `proc_exit` doesn't access memory, take any strings, use any handles, etc., so it doesn't need any of the wig machinery. It can just be a plain `extern "C"` function somewhere, and different runtimes can still do different things in the implementation -- panic as Lucet does, or trigger a trap-like unwind as this PR does. And that way, it can trap without having to worry about unwinding through wig's Rust code.